src/routes/admin/domains/domain.ts

```diff
@@ -4,35 +4,64 @@
  * Author : Madhavi Losetty
  **********************************************************************/
 import { check } from 'express-validator'
+import { NodeForge } from '../../../NodeForge'
+import { CertManager } from '../../../CertManager'
+import Logger from '../../../Logger'
 
 export const domainInsertValidator = (): any => {
+  const nodeForge = new NodeForge()
+  const certManager = new CertManager(new Logger('CertManager'), nodeForge)
+  let pfxobj
   return [
     check('profileName')
       .not()
       .isEmpty()
       .withMessage('AMT Domain profile name is required')
       .matches('^[a-zA-Z0-9$@$!%*#?&-_~^]+$')
       .withMessage('AMT Domain profile name accepts letters, numbers, special characters and no spaces'),
+    check('provisioningCertPassword')
+      .not()
+      .isEmpty()
+      .withMessage('Provisioning Cert Password is required')
+      .isLength({ max: 64 })
+      .withMessage('Password should not exceed 64 characters in length')
+      .custom((value, { req }) => {
+        pfxobj = certManager.convertPfxToObject(req.body.provisioningCert, req.body.provisioningCertPassword)
```

Comment:
What's the message that will be returned if the password is incorrect? Will it tell them to check their password?

Reply:
I'm thinking that's most relevant to the user for the API, however, it could fail for other reasons right? Such as the cert not being correctly formed.  So it seems that the the current error message is the right message as an internal error, but I'm thinking a try/catch around this call to send something more user friendly that indicates what could go wrong and what to do. Something like "Unable to decrypt provisioning certificate. Please check that the password is correct, and that the certificate is a valid certificate."